src/helpers/ControlRecordTreeMgr.js

```diff
@@ -1,31 +1,37 @@
 import { Util } from './Util';
 import sortBy from 'lodash/sortBy';
+import { getUpdatedFormFieldPath } from 'src/helpers/formNamespace';
 
 export default class ControlRecordTreeMgr {
 
-  generateNextTree(rootTree, formFieldPath) {
+  generateNextTree(rootTree, formFieldPath, parentFormFieldPath) {
     let updatedTree = this.getLatestBrotherTree(rootTree, formFieldPath);
+    const updatedFormFieldPath = parentFormFieldPath === undefined ?
+        this.getNextFormFieldPath(formFieldPath)
+        : getUpdatedFormFieldPath(updatedTree, parentFormFieldPath);
+    updatedTree = updatedTree.set('formFieldPath', updatedFormFieldPath);
     if (updatedTree.children && updatedTree.children.size > 0) {
       const filteredTree = this.filterChildTree(updatedTree);
       const clonedChildTree = filteredTree.map(r => (
-                this.generateNextTree(rootTree, r.formFieldPath)
+                this.generateNextTree(rootTree, r.formFieldPath, updatedTree.formFieldPath)
             ));
       updatedTree = updatedTree.set('children', clonedChildTree);
     }
-    const nextFormFieldPath = this.generateFormFieldPath(updatedTree);
-    return updatedTree.set('formFieldPath', nextFormFieldPath).set('value', {}).set('active', true);
+    return updatedTree.set('value', {}).set('active', true);
   }
 
   filterChildTree(updatedTree) {
```

Comment:
@shivarach can you please explain what are we doing as part of **filterChildTree** method ??

Reply:
We are filtering out any child created using add more.